crates/sui-framework/packages/sui-framework/sources/hanger.move

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Hanger is an object that stores internally a given type T. While Hanger is serializable via bcs::to_bytes, its
+/// internal type is not directly serializable (as long as the caller does not expose the internal type).
+module sui::hanger {
```

Comment:
We are discussing changes to private `entry` rules. So while this is cute, I _really_ hope we don't have to land this.

Reply:
Then again, the functionality does seem maybe somewhat useful? Though I'm not sure I'm in love with the name :P